File: Cs/Solution/UniqueMorseRepresentations.cs

using System.Collections.Generic;

namespace Solution
{
	public partial class Solution
	{
		public int UniqueMorseRepresentations(string[] words)
		{
			string[] morse = new string[] { ".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.." };
			HashSet<string> set = new HashSet<string>();
			foreach (string word in words)
			{
				string str = "";
				foreach (char c in word)
				{
					str += morse[c - 'a'];
				}
				set.Add(str);
			}
			return set.Count;
		}
	}
}

File: Ruby/solution/unique_morse_representations.rb
# @param {String[]} words
# @return {Integer}
def unique_morse_representations(words)
  morse = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---",
           "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-",
           "..-", "...-", ".--", "-..-", "-.--", "--.."]
  words.map { |w| w.chars.map { |c| morse[c.ord - "a".ord] }.join }.uniq.size
end
